x-pack/plugins/enterprise_search/server/lib/indices/fetch_ml_inference_pipeline_processors.ts

```diff
@@ -149,8 +164,13 @@ export const fetchAndAddTrainedModelData = async (
 
 export const fetchMlInferencePipelineProcessors = async (
   client: ElasticsearchClient,
+  trainedModelsProvider: MlTrainedModels | undefined,
   indexName: string
 ): Promise<InferencePipeline[]> => {
+  if (!trainedModelsProvider) {
+    return Promise.reject(new Error('Machine Learning is not enabled'));
+  }
```

Comment:
I found that if I disable ML in a Kibana space, I get an access error before reaching this line (this logic is already part of the codebase):
```
[2022-10-14T11:36:36.278-04:00][ERROR][plugins.enterpriseSearch] Error: Insufficient privileges to access feature
```
Therefore I won't add any explicit handler code to the API (`indices.ts`) for this specific use case - if we ever reach it, it will be treated as a generic server error.